treeherder/seta/job_priorities.py

```diff
@@ -0,0 +1,136 @@
+    '''
+    Sample data from Treecodes().jobnames_query() (skipping irrelevant fields)
+    {
+        "buildplatform": "buildbot",
+        "buildtype": "debug",
+        "name": "web-platform-tests-1",
+        "platform": "windows8-64",
+        "ref_data_name": "Windows 8 64-bit mozilla-inbound debug test web-platform-tests-1",
+    },{
+        "buildplatform": "taskcluster",
+        "buildtype": "opt",
+        "name": "mochitest-webgl-e10s-1",
+        "platform": "linux64",
+        "ref_data_name": "desktop-test-linux64/opt-mochitest-webgl-e10s-1",
+    }
+    '''
+    map = {}
+    for job in Treecodes().jobnames_query():
+        if job['buildplatform'] == build_system:
+            key = unique_key(testtype=job['name'],
+                             buildtype=job['buildtype'],
+                             platform=job['platform'])
+            map[key] = job['ref_data_name']
+
+    return map
+
+
+def _process(build_system, job_priorities):
+    '''Return list of ref_data_name for job_priorities'''
+    jobs = []
+    # This map contains the ref_data_name of every Treeherder job
+    map = _map(build_system)
+
+    for jp in job_priorities:
+        if build_system == 'taskcluster':
+            # We need to retranslate the jobtype back to the proper data form after all.
+            # XXX: Ask jmaher, what would it take to fix this?
+            jp.testtype = jp.testtype.replace('e10s-browser-chrome', 'browser-chrome-e10s')
+            jp.testtype = jp.testtype.replace('e10s-devtools-chrome', 'devtools-chrome-e10s')
+            jp.testtype = jp.testtype.replace('gl-', 'webgl-')
+
+        key = jp.unique_identifier()
+        if key in map:
+            # e.g. desktop-test-linux64-pgo/opt-reftest-13 or builder name
+            jobs.append(map[key])
+        else:
+            LOG.warning('We did not find job priority ({}) in the list of accepted jobs'.format(jp))
+
+    return jobs
+
+
+def _gecko_decision_task_request(project):
+    # In the case of a request from the Gecko decision task we don't care which priority is
+    # requested. We return either all jobs every 5th push or 90 minutes for that project or
+    # all TaskCluster high value jobs.
+    task_request = _update_task_request(project)
+    last_reset = task_request.seconds_since_last_reset()
+    job_priorities = []
+    for jp in JobPriority.objects.all().exclude(buildsystem='buildbot'):
+        # We only add a job if it hasn't reached its timeout.
+        # If the timeout is zero it means that the job needs to always run
+        # XXX Ask jmaher to evaluate this condition
+        if last_reset < jp.timeout or jp.timeout == 0:
+            # Due to the priority of all high value jobs is 1, and we
+            # We need to return all jobs for every 5th push.
+            if task_request.counter % jp.priority != 0:
+                job_priorities.append(jp)
+
+    return _process(build_system='taskcluster', job_priorities=job_priorities)
```

Comment:
I am fine with the counter logic, just not sure about build_system='*'.  Right now we return _process(build_system='taskcluster'...), will that get all taskcluster tasks, or only ones with build_system=='taskcluster' and ignore ones where build_system=='*' ?